components/patina_smbios/src/manager/core.rs

```diff
@@ -113,9 +116,79 @@ impl SmbiosManager {
             minor_version,
             entry_point_64: RefCell::new(None),
             table_64_address: RefCell::new(None),
+            // Pre-allocated buffers start as None, set up during component init
+            table_buffer_addr: RefCell::new(None),
+            table_buffer_max_size: SIZE_64KB,
+            ep_buffer_addr: RefCell::new(None),
         })
     }
 
+    /// Allocate buffers for SMBIOS table publication
+    ///
+    /// Allocates buffers for the SMBIOS table and entry point upfront to avoid
+    /// repeated allocations. The buffers are reused for all table publications.
+    ///
+    /// Also adds the Type 127 End-of-Table marker to maintain the invariant that
+    /// it's always the last record.
+    ///
+    /// # Errors
+    ///
+    /// Returns `SmbiosError::AllocationFailed` if buffer allocation fails.
+    /// Returns `SmbiosError::HandleExhausted` if unable to allocate handle for Type 127.
+    ///
+    /// ## Coverage Exclusion
+    ///
+    /// This function is excluded from unit test coverage (`#[coverage(off)]`) because testing it in
+    /// unit tests would require significant test infrastructure:
+    /// 1. PageAllocation requires a `&'static MemoryManager` reference (requires Box::leak or unsafe transmute)
+    /// 2. Mock MemoryManager would need to allocate real memory and handle free_pages() on drop
+    /// 3. Test would need to manage memory lifecycle without UEFI page allocator
```

Comment:
nitpick: I appreciate the detail, but I think there's a danger of it becoming obsolete over time if details of memory allocation change. I think it would be sufficient to say something like "testing it in unit tests would require test infrastructure to mock memory allocations"

Also, looking at the implementation, I think you'd just instantiate a `MockMemoryManager` and drop a couple of expectations on it for two `allocate_pages()` calls (though they would have to make allocations to meet the caller expectations).